ts/Stock/Indicators/TrendLine/TrendLineIndicator.ts

```diff
@@ -106,14 +107,21 @@ class TrendLineIndicator extends SMAIndicator {
         series: TLinkedSeries,
         params: TrendLineParamsOptions
     ): IndicatorValuesObject<TLinkedSeries> {
-        const xVal: Array<number> = (series.xData as any),
+        const orgXVal: Array<number> = (series.xData as any),
             yVal: Array<Array<number>> = (series.yData as any),
             LR: Array<Array<number>> = [],
             xData: Array<number> = [],
             yData: Array<number> = [],
-            xValLength: number = xVal.length,
+            xValLength: number = orgXVal.length,
             index: number = (params.index as any);
 
+        let xVal: Array<number> = [];
+        // Create a fake xVal array with consecutive values to avoid issues with
+        // ordinal, point.remove etc. #18710
+        for (let i = 0; i < xValLength; i++) {
+            xVal[i] = i;
+        }
```

Comment:
Thank you for the improvement suggestions, it really simplified the whole solution!

For the `sumX2` i decided to stick with the `i * i` approach, since applying the sum of squares of “n” natural numbers algorithm won't work here, because 0 (the first element of the fake array that we still have to keep in mind, even if it's not created here) is not a natural number. Or maybe I'm wrong?

Reply:
No, you are right. But we can improve that equation a little: instead of `len`, we can use `len - 1` (to skip zero). Then we will have `((xValLength -1) * (xValLength) * (2 * xValLength - 1)) / 6`

Tested with: https://jsfiddle.net/BlackLabel/wL2h739o/1/

Reply:
Thank you, that's exactly what we wanted!